src/optimal_image/search.rs

```diff
@@ -0,0 +1,58 @@
+use dataclients::ImageData;
+use std::path::Path;
+use ImageConfig;
+use ImageFormat;
+use ImageScore;
+
+pub struct SearchOptions {
+    threshold: f64,
+    formats: Vec<ImageFormat>,
+}
+
+pub struct Search {
+    image_data: ImageData,
+    options: SearchOptions,
+}
+
+impl Search {
+    fn from_path(&self, path: &Path) -> &Search {
+        // TODO: initialize struct members based on path
+        self
+
+
+
+    }
+}
+
+pub fn find_optimal_config(
+    config: &Vec<ImageConfig>,
+    threshold: f64,
+) -> &ImageConfig {
+    let index = config
+        .binary_search_by(|probe| {
+            // TODO:
+            // generate image with `probe` config
+            // run dssim against original
+            // return Ordering according to threshold
+            unimplemented!();
+        })
+        .unwrap();
```

Comment:
Unless we are sure that this will always wrap to a value, lets try avoiding this as it will lead to a panic